### PR TITLE
Support Project Https Protocol Pings

### PR DESCRIPTION
**Pull request by maysunfaisal**
Signed-off-by: Maysun J Faisal <maysun.j.faisal@ibm.com>

Issue https://github.com/eclipse/codewind/issues/269

This PR enables Turbine layer to ping a project successfully if it's https enabled. This will allow the Codewind IDE plugins to open the application over https on the browser.

This feature is enabled by setting the boolean property `isHttps` in `.cw-settings`

Tested on both Local & Kube. Also updated the Portal cw-settings tests.

IDE now reports an https project as running in the overview page:
Screen Shot 2019-08-16 at 5 50 38 PM

Here is my python application over https in the browser:
Screen Shot 2019-08-16 at 5 50 44 PM

### CODE REVIEW

**Comment by maysunfaisal** [comment]
@rajivnathan @stalleyj can you please help review the PR? Thx!

**Comment by stalleyj** [comment]
@maysunfaisal The various settings are only supposed to be in either .cwSettings or the inf to prevent them getting out of sink, link contextRoot or healthcheck. I thought the IDE's would get those values from the settings file @tetchel how does the plugin get the value of contextroot ?

**Comment by maysunfaisal** [comment]
@stalleyj The IDE wanted the essential project settings key to be included in .inf files.. They are updated whenever a setting value changes and during project creation. You can find this discussion from couple of months back in Portal issue 1053 and has been in the product for close to over a month now
@tetchel

**Comment by maysunfaisal** [comment]
@stalleyj thx!

**Comment by tetchel** [comment]
Maysun is correct, the IDEs only pick up project data such as contextRoot from the inf (`api/v1/projects`).

File: src/pfe/file-watcher/server/src/projects/projectUtil.ts [hunk 1]
```diff
@@ -966,15 +1011,24 @@ export async function isApplicationUp(projectID: string, handler: any): Promise<
             updateDetailedAppStatus(projectID, containerInfo.ip, port, path);
         }
 
+        const isHttps: boolean = await getProjectWWWProtocolMap(projectID);
```
Comment (rajivnathan) [inline src/pfe/file-watcher/server/src/projects/projectUtil.ts]: Do we need the `getProjectWWWProtocolMap()` function? Shouldn't `projectInfo` contain the `isHttps` property? If we don't need that function we don't need the `projectWWWProtocolMap` either.

File: src/pfe/file-watcher/server/src/projects/projectUtil.ts [hunk 2]
```diff
@@ -901,6 +907,44 @@ export async function isContainerActive(projectID: string, handler: any): Promis
     }
 }
 
+// /**
```
Comment (rajivnathan) [inline src/pfe/file-watcher/server/src/projects/projectUtil.ts]: This commented code should be removed.

**Pull request merged**